components/kubearchive/base/migration-job.yaml

```
@@ -0,0 +1,40 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: kubearchive-schema-migration
+  annotations:
+    ignore-check.kube-linter.io/no-read-only-root-fs: >
+      "This job needs to clone a repository to do its job, so it needs write access to the FS."
+spec:
+  parallelism: 1
+  backoffLimit: 4
+  template:
+    spec:
+      containers:
+        - name: migration
+          image: quay.io/fedora/python-311:20240911
+          command: ["/bin/sh", "-c"]
+          args:
+            - git clone https://github.com/kubearchive/kubearchive --depth=1 --branch=${KUBEARCHIVE_VERSION} /tmp/kubearchive;
+              cd /tmp/kubearchive;
+              curl -L https://github.com/golang-migrate/migrate/releases/download/${MIGRATE_VERSION}/migrate.linux-amd64.tar.gz | tar xvz migrate;
+              ./migrate -verbose -path integrations/database/postgresql/migrations/ -database postgresql://${DATABASE_USER}:${DATABASE_PASSWORD}@${DATABASE_URL}:${DATABASE_PORT}/${DATABASE_DB} up
```

```suggestion
            - git clone https://github.com/kubearchive/kubearchive --depth=1 "--branch=${KUBEARCHIVE_VERSION}" /tmp/kubearchive;
              cd /tmp/kubearchive || exit 1;
              curl -L "https://github.com/golang-migrate/migrate/releases/download/${MIGRATE_VERSION}/migrate.linux-amd64.tar.gz" | tar xvz migrate;
              ./migrate -verbose -path integrations/database/postgresql/migrations/ -database "postgresql://${DATABASE_USER}:${DATABASE_PASSWORD}@${DATABASE_URL}:${DATABASE_PORT}/${DATABASE_DB}" up
```

suggestions from shellcheck

---

Thanks! However I'll pass on the suggestions.